From 39f127ea4fb3f287fe84e8b68ff0d3675f90506e Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Fri, 30 May 2014 11:35:50 -0700
Subject: [PATCH donotupstream] arm, vt8500, LLVMLlinux: Comment out mcr if used with clang

This is a workaround patch.

arch/arm/mach-vt8500/vt8500.c:72:11: error: invalid % escape in inline assembly string
        asm("mcr%? p15, 0, %0, c7, c0, 4" : : "r" (0));
            ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.


Not-Signed-off-by: Behan Webster <behanw@converseincode.com>

---
 arch/arm/mach-vt8500/vt8500.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux.orig/arch/arm/mach-vt8500/vt8500.c
+++ linux/arch/arm/mach-vt8500/vt8500.c
@@ -69,7 +69,9 @@
 {
 	local_irq_disable();
 	writew(5, pmc_base + VT8500_HCR_REG);
+#ifndef __clang__
 	asm("mcr%? p15, 0, %0, c7, c0, 4" : : "r" (0));
+#endif
 }
 
 void __init vt8500_init(void)
